### PR TITLE
Update CONTRIBUTING.md

### LINKED ISSUE #3170: Update CONTRIBUTING.md
**Is your feature request related to a problem? Please describe.**
Our Contributing Guidelines are currently empty.

**Describe the solution you'd like**
Update and fill our Contributing Guidelines properly. Reference our coding style for C#.

### PR DESCRIPTION
**Pull request by AntiD2ta**
Resolves #3170 

## Changes:
- Update our contributing guidelines

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [ ] No

## Further comments

Help needed in the _Commit Messages_ and _Branch Naming_ sections.

Requires [this](https://github.com/NethermindEth/docs/pull/51) to be merged.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1 +1,74 @@
...
+
+### Improving Issues and PR
+
+Please add, if possible, a reviewer, assignees and labels to your issue and PR.
+  
+## DOs and DON'Ts
+
+Please do:
+
+* **DO** follow our [coding style](https://docs.nethermind.io/nethermind/guides-and-helpers/coding-style). (C# code-specific)
```
Comment (1swaraj) [inline CONTRIBUTING.md:35]: Remove this because it is not ready as of now.

**Review by 1swaraj — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (AntiD2ta) [inline CONTRIBUTING.md:35]:
```suggestion
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1 +1,73 @@
...
+  
+Please do not:
+
+* **DON'T** create a new file without the proper file header.
+* **DON'T** fill the issues and PR descriptions vaguely. The elements in the templates are there for a good reason. Help the team. 
+* **DON'T** surprise us with big pull requests. Instead, file an issue and start a discussion so we can agree on a direction before you invest a large amount of time.
+
+## Commit Messages
+
+[This is a good opportunity to create a convention for commit messages]
```
Comment (matilote) [inline CONTRIBUTING.md:47]: Do we leave it as is? Same for branch naming?

Comment (AntiD2ta) [reply CONTRIBUTING.md:47]: I will open a discussion with eth team to get to a consensus about this

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1 +1,69 @@
...
+## Branch Naming
+
+Branch names must follow `snake_case` pattern. Follow the pattern `feature/<name>` or `fix/<name>` `(folder/<name>)` when it is possible and add issue reference if applicable.
+
+## File Headers
+
+The following file header is the used for Nethermind. Please use it for new files.
+
+```
+//  Copyright (c) 2021 Demerzel Solutions Limited
```
Comment (AntiD2ta) [inline CONTRIBUTING.md:54]:
```suggestion
//  Copyright (c) 2022 Demerzel Solutions Limited
```

**Review by matilote — APPROVED** [review]

**Pull request merged**